activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -244,6 +244,22 @@ def assert_no_changes(expression, message = nil, from: UNTRACKED, &block)
 
         retval
       end
+
+      private
+        def _assert_nothing_raised_or_warn(assertion, &block)
+          assert_nothing_raised(&block)
+        rescue Minitest::UnexpectedError => e
+          if tagged_logger && tagged_logger.warn?
```

Comment:
Won't this result in a
```
got #<NameError: undefined local variable or method `tagged_logger'
```
if you don't explicitly set `self.tagged_logger =`, @ghiculescu ?
WDYT of `defined?(tagged_logger) && tagged_logger&.warn?` instead?

Where is `tagged_logger` normally assigned to be accessible in this context?

Reply:
I think I copied this from https://github.com/rails/rails/blob/18707ab17fa492eb25ad2e8f9818a320dc20b823/activesupport/lib/active_support/testing/tagged_logging.rb#L11

That module is included here https://github.com/rails/rails/blob/18707ab17fa492eb25ad2e8f9818a320dc20b823/activesupport/lib/active_support/test_case.rb#L129

I can change to a `defined?` check if needed, but curious how you have it set up that it's breaking for you?

Reply:
We should not change it. Rails is assuming correctly that this method exists because it does in all `ActiveSupport::TestCase`. This problem only happens in RSpec, so RSpec should be responsible to makes sure that method exist.